sn_networking/src/lib.rs

```diff
@@ -704,6 +708,13 @@ impl Network {
                             && target_record == Some(returned_record.clone()))
                     {
                         return Ok(returned_record);
+                    } else if expect_no_prior_record
+                        && (target_record.is_some()
+                            && target_record != Some(returned_record.clone()))
+                    {
+                        error!("Record check: expected {target_record:?}");
+                        error!("Returned: {returned_record:?}");
+                        return Err(Error::ExpectedNoRecordToExist(returned_record.key.into()));
```

Comment:
Should we also return error if `target_record.is_none()`? 
Also can we add some docs for this argument?